File: 算法/代码随想录算法PDF/00调试源代码/1047.删除字符串中的所有相邻重复项.py
#
# @lc app=leetcode.cn id=1047 lang=python3
#
# [1047] 删除字符串中的所有相邻重复项
#

# @lc code=start
class Solution:
    def removeDuplicates(self, s: str) -> str:
        s = list(s)
        slow, fast = 0, 0
        while fast < len(s):
            s[slow] = s[fast]
            if slow > 0 and s[slow] == s[slow - 1]:
                slow -= 1
            else:
                slow += 1
            fast += 1
        return ''.join(s[:slow])
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/1047.删除字符串中的所有相邻重复项.cpp
/*
 * @lc app=leetcode.cn id=1047 lang=cpp
 *
 * [1047] 删除字符串中的所有相邻重复项
 */

// @lc code=start
class Solution {
public:
    string removeDuplicates(string s) {
        int slow = 0, fast = 0;
        while (fast < s.size()) {
            s[slow] = s[fast];
            if (slow > 0 && s[slow] == s[slow - 1]) {
                slow--;
            } else {
                slow++;
            }
            fast++;
        }
        return s.substr(0, slow);
    }
};
// @lc code=end

